src/Sys/InspectSeoHandler.php

```diff
@@ -0,0 +1,176 @@
+    {
+        if ($this->flags->getBoolean('inspect_seo_handler_google_disabled')) {
+            return $command;
+        }
+
+        $request = new \Google_Service_SearchConsole_InspectUrlIndexRequest();
+        $siteUrl = $this->flags->getString('inspect_seo_google_site_url');
+
+        if (empty($siteUrl)) {
+            $this->logger->error('Must provide a site url.', [
+                'site_url' => $siteUrl,
+            ]);
+
+            return $command;
+        }
+
+        $request->setSiteUrl($siteUrl);
+        $url = UriTemplateService::expand(
+            "{$node::schema()->getQName()}.canonical", $node->getUriTemplateVars()
+        );
+        $request->setInspectionUrl($url);
+        $client = new \Google_Client();
+        $client->addScope(\Google_Service_SearchConsole::WEBMASTERS_READONLY);
+
+        $response = null;
+
+        try {
+            $base64EncodedAuthConfig = Crypto::decrypt(getenv('GOOGLE_SEARCH_CONSOLE_API_SERVICE_ACCOUNT_OAUTH_CONFIG'), $this->key);
+            $client->setAuthConfig(json_decode(base64_decode($base64EncodedAuthConfig), true));
+            $service = new \Google_Service_SearchConsole($client);
+            $response = $service->urlInspection_index->inspect($request);
+        } catch (\Throwable $e) {
+            $this->logger->error('An error occurred in checkIndexStatus for google.', [
+                'exception' => $e,
+                'node_ref' => $node->generateNodeRef(),
+                'url' => $url,
+                'retry_count' => $command->get('ctx_retries'),
+                'stack_trace' => $e->getTraceAsString(),
+            ]);
+        }
+
+        if ($response === null) {
+            return $command;
+        }
+
+        $inspectionResult = $response->getInspectionResult();
+        $indexStatusResult = $inspectionResult->getIndexStatusResult() ?? null;
+        $ampResult = $inspectionResult->getAmpResult() ?? null;
+
+        $isConclusiveForWeb = $indexStatusResult && ($indexStatusResult->getVerdict() === 'PASS' || $indexStatusResult->getVerdict() === 'FAIL');
+        $isConclusiveForAmp = !$node::schema()->hasField('amp_enabled') || ($ampResult && ($ampResult->getVerdict() === 'PASS' || $ampResult->getVerdict() === 'FAIL'));
+
+        if ($isConclusiveForWeb && $isConclusiveForAmp) {
+            $this->putEvent($command, $pbjx, $node, 'google', $response);
+            return $command;
+        }
+
+        $retries = $command->get('ctx_retries');
+        $maxRetries = $this->flags->getInt('inspect_seo_max_retries', 5);
+
+        if ($retries < $maxRetries) {
+            $command->addToSet('search_engines', ['google']);
+            return $command;
+        }
+
+        $this->putEvent($command, $pbjx, $node, 'google', $response);
+
+        return $command;
+    }
+
+
+    private function putEvent(Message $command, Pbjx $pbjx, Message $node, string $searchEngine, InspectUrlIndexResponse $response): void
```

Comment:
@efjacobson @digitalartcrew do you actually want this on the node's event stream or just want to publish it?  if it going to be on the event stream for a node (e.g. an article) then you want to use the aggregate resolver, and add methods for the command and the event (apply phase).  without it your aggregate will run the events, fail to find a applySeoInspected method and fail.

Reply:
@gdbrown good point, no i suppose we only really need to publish it

Reply:
@efjacobson does anything on the article need to be updated when an inspection occurs?

Reply:
@gdbrown nope. it does not need to be put on the node's stream, just published so it can be subscribed to elsewhere